includes/payment-methods/class-wc-stripe-payment-request.php

```diff
@@ -1129,40 +1093,55 @@ public function ajax_get_selected_product_data() {
 	public function ajax_add_to_cart() {
 		check_ajax_referer( 'wc-stripe-add-to-cart', 'security' );
 
-		if ( ! defined( 'WOOCOMMERCE_CART' ) ) {
-			define( 'WOOCOMMERCE_CART', true );
-		}
+		try {
+			if ( ! defined( 'WOOCOMMERCE_CART' ) ) {
+				define( 'WOOCOMMERCE_CART', true );
+			}
 
-		WC()->shipping->reset_shipping();
+			$product_id       = isset( $_POST['product_id'] ) ? absint( $_POST['product_id'] ) : 0;
+			$product          = wc_get_product( $product_id );
+			$post_input       = filter_input_array( INPUT_POST, [ 'shipping_pending' => FILTER_VALIDATE_BOOLEAN ] );
+			$shipping_pending = $post_input['shipping_pending'];
+
+			if ( ! is_a( $product, 'WC_Product' ) ) {
+				/* translators: %d is the product Id */
+				throw new Exception( sprintf( __( 'Product with the ID (%d) cannot be found.', 'woocommerce-gateway-stripe' ), $product_id ) );
+			}
 
-		$product_id   = isset( $_POST['product_id'] ) ? absint( $_POST['product_id'] ) : 0;
-		$qty          = ! isset( $_POST['qty'] ) ? 1 : absint( $_POST['qty'] );
-		$product      = wc_get_product( $product_id );
-		$product_type = $product->get_type();
+			$qty          = ! isset( $_POST['qty'] ) ? 1 : absint( $_POST['qty'] );
+			$product_type = $product->get_type();
 
-		// First empty the cart to prevent wrong calculation.
-		WC()->cart->empty_cart();
+			if ( ! $product->has_enough_stock( $qty ) ) {
+				/* translators: 1: product name 2: quantity in stock */
+				throw new Exception( sprintf( __( 'You cannot add that amount of "%1$s"; to the cart because there is not enough stock (%2$s remaining).', 'woocommerce-gateway-stripe' ), $product->get_name(), wc_format_stock_quantity_for_display( $product->get_stock_quantity(), $product ) ) );
+			}
 
-		if ( ( 'variable' === $product_type || 'variable-subscription' === $product_type ) && isset( $_POST['attributes'] ) ) {
-			$attributes = wc_clean( wp_unslash( $_POST['attributes'] ) );
+			// Do we need to do this before emptying the cart?
```

Comment:
> Do we need to do this before emptying the cart?

I don't know the answer for this question either, but apparently from WC's PHPDoc, this function is used to "Reset the totals for shipping as a whole.".

I usually tend to ignore or forget about comments. Let's not forget this comment as a question and add a "TODO" in front of it, or resolve this question and change the comment.

Reply:
This was a TODO for me. I forgot to remove it. And yes. I found out later this line is necessary. I'll remove this comment

Reply:
done b5dadd32f1bf73de0abfe4170ee4f24562be9de9